### PR TITLE
Dynamic install of plugins

### PR DESCRIPTION
**Pull request by tupe12334**
This PR adds the logic of dynamically installation of plugins in the generate-code DSG executer

### CODE REVIEW

File: packages/amplication-data-service-generator/src/generate-code.ts [hunk 1]
```diff
@@ -19,14 +18,28 @@ generateCode(source, destination).catch((err) => {
...
+    const resourceData = await readInputJson(source);
+    const { pluginInstallations } = resourceData;
+
+    console.log("Start code generation dynamic plugins installation");
+    await dynamicPluginInstallation(
+      pluginInstallations.map((plugin) => plugin.npm)
+    );
+    console.log("Finish code generation dynamic plugins installation");
+
+    const modules = await (
```
Comment (barshimi) [inline packages/amplication-data-service-generator/src/generate-code.ts:40]: why don't you import the file before the run and use it on invoke ?

File: packages/amplication-data-service-generator/src/tests/pluginInstallation.ts [hunk 2]
```diff
@@ -1,3 +1,3 @@
 import { PluginInstallation } from "@amplication/code-gen-types";
-
-export const installedPlugins: PluginInstallation[] = [];
+import { plugins } from "../../scripts/constants/example-plugins";
```
Comment (barshimi) [inline packages/amplication-data-service-generator/src/tests/pluginInstallation.ts:2]: again

File: packages/amplication-data-service-generator/src/dynamic-plugin-installation.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    child.addListener("exit", resolve);
+  });
+}
+
+export async function dynamicPluginInstallation(
+  pluginInstallations: string[]
+): Promise<void> {
+  await promiseFromChildProcess(
+    exec(`npm i ${pluginInstallations.join(" ")} --no-save -f`)
+  );
```
Comment (overbit) [inline packages/amplication-data-service-generator/src/dynamic-plugin-installation.ts:15]: This approach might bring us easily into a dependency hell https://npm.github.io/how-npm-works-docs/theory-and-design/dependency-hell.html.

Plus is locking our solution to only support NPM leaving most used package managers like yarn and pnpm out to support. 
Possibly, a stronger solution would be to bundle all the dependencies of a plugin within the plugin itself.

**Review by overbit — CHANGES_REQUESTED** [review]

File: packages/amplication-data-service-generator/src/dynamic-plugin-installation.ts (hunk 3 above)
Comment (overbit) [reply packages/amplication-data-service-generator/src/dynamic-plugin-installation.ts:15]: Plus we might have network issues (that are always a possibility) that will affect the installation of npm packages in runtime and, consequently, the code generation.

**Comment by GreenMachine01** [comment]
@tupe12334 Please see feedback and resolve conflicts.

**Pull request merged**